include/sdf/system_util.hh

```diff
@@ -29,5 +29,16 @@
  */
 #define SDFORMAT_HIDDEN GZ_SDFORMAT_HIDDEN
 
+/** \def SDFORMAT_WARN_UNUSED
+ * Use to represent "[[warn_if_unused]]" compiler directive on C++17 or above
+ */
+
+#if __cplusplus >= 201703L
+#define SDFORMAT_WARN_UNUSED [[nodiscard]]
+#else
+#error "HELLO"
+#define SDFORMAT_WARN_UNUSED
+#endif
```

Comment:
per https://github.com/gazebosim/sdformat/issues/1358#issuecomment-1916778503, we don't need to define this macro, we can use `[[nodiscard]]` directly in the header file. apologies for the back-and-forth

Reply:
No worries. While I think I could do this to the model, this is perhaps a great candidate for using clang-tidy's automated refactoring capabilities. Although I don't have experience with it, perhaps someone at Gazebo does? 

I don't really expect to have time to do it to the entire repo, at least manually. It's not quick work because you can't find-replace or regex it.

Reply:
I haven't configured `clang-tidy` before. I think it would be worth adding `[[nodiscard]]` to this class by hand and then comparing to what clang-tidy generates

Reply:
I installed `clang-tidy` on a debian machine and built Ionic up to `sdformat15` in a colcon workspace and then was able to apply the `[[nodiscard]]` attribute automatically using the following command after sourcing the workspace:

~~~
clang-tidy --fix -checks=-*,modernize-use-nodiscard src/sdformat/include/sdf/*.hh --  $(pkg-config --cflags sdformat15)
~~~

Reply:
Fantastic. I'll go ahead and do the work manually and post a diff.

Reply:
untested changes to all header files in https://github.com/gazebosim/sdformat/commit/1d96a587497de7705b3cf2840b2ab2a8306d75fb

I'm pretty sure it pushes some lines over 80 characters, which would need to be fixed